CMakeLists.txt

```diff
@@ -69,6 +69,11 @@ add_subdirectory(dstulib)
 add_subdirectory(engine)
 add_subdirectory(keylib)
 
+target_include_directories(dstulib PRIVATE ${OPENSSL_INCLUDE_DIR})
```

Comment:
Please use imported target `OpenSSL::Crypto` and move this change to the `dstulib/CMakeLists.txt`.

Reply:
Moved this line to `dstulib/CMakeLists.txt`.

But if I replace `${OPENSSL_INCLUDE_DIR}` with `OpenSSL::Crypto` then it fails with error:

```
dstu-engine/dstulib/asn1.h:9:10: fatal error: 'openssl/asn1.h' file not found
```

Reply:
Interesting. Could you please provide more compilation log for this error?

Reply:
If I add this line to `dstulib/CMakeLists.txt`
```
target_include_directories(dstulib PRIVATE OpenSSL::Crypto)
```

then build fails:
```
~/Projects/dstu-engine/build$ cmake ..
-- The C compiler identification is AppleClang 16.0.0.16000026
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Found OpenSSL: /opt/homebrew/Cellar/openssl@3/3.3.2/lib/libcrypto.dylib (found suitable version "3.3.2", minimum required is "1.1.0")
-- Second argument of PKEY copy method is constant.
-- Second argument of PKEY ASN.1 decode method is constant.
-- Found PkgConfig: /opt/homebrew/bin/pkg-config (found version "0.29.2")
-- Performing Test Iconv_IS_BUILT_IN
-- Performing Test Iconv_IS_BUILT_IN - Failed
-- Found Iconv: /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/lib/libiconv.tbd
-- Configuring done (1.2s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/strump/Projects/dstu-engine/build


~/Projects/dstu-engine/build$ make
[  4%] Building C object dstulib/CMakeFiles/dstulib.dir/key.c.o
In file included from /Users/strump/Projects/dstu-engine/dstulib/key.c:6:
In file included from /Users/strump/Projects/dstu-engine/dstulib/key.h:9:
/Users/strump/Projects/dstu-engine/dstulib/asn1.h:9:10: fatal error: 'openssl/asn1.h' file not found
    9 | #include <openssl/asn1.h>
      |          ^~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [dstulib/CMakeFiles/dstulib.dir/key.c.o] Error 1
make[1]: *** [dstulib/CMakeFiles/dstulib.dir/all] Error 2
make: *** [all] Error 2
```

On my machine variable `OPENSSL_INCLUDE_DIR` points to `/opt/homebrew/Cellar/openssl@3/3.3.1/include`

Reply:
Actually this line is not in the root `CMakeList.txt` but in the `dstulib/CMakeLists.txt`.

I tried to move it back to the root `CMakeList.txt`.

* If I put it before the `add_subditrectory(dstulib)` then CMake fails with `Cannot specify include directories for target "dstulib" which is not built`
* If I put it after - compilation fails with `dstu-engine/dstulib/asn1.h:9:10: fatal error: 'openssl/asn1.h' file not found`.

Did you try to add `target_include_directories(...)` in your environment? Does it work on Linux? Could it be CMake version issue?

Reply:
Imported targets should work as a replacement for both `target_link_libraries` and `target_include_directories`. I prefer use them as a more uniform and modern way of managing dependencies.

Let's put it aside for some time, I'll dust off my old MBP and try your branch on OSX for myself.

Meanwhile, what is your CMake version?

Reply:
I'm using CMake version 3.30.4

Code from this branch works pirfectly for me. I can build `dsto.so` and use at as OpenSSL engine to handle DSTU 4145-2002 encryption. If you see any room for improvements, I'm open for suggestions.

Reply:
Please see my changes in #25 

You can try them in the `fix-osx-build` branch.

If my solution works for you, I'll merge it.